pkg/kopia/repository/connect_to_or_create_repository.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 The Kanister Authors.
```

Comment:
Not a fan of the file names in this package. Since it already has `repository` in the package name, I think the following sound better:
- connect_server.go
- create_or_connect.go
- create.go
- connect.go

I don't mind having all these helpers in the same file - `repository.go` either 🤔 

@ankitjain235 @PrasadG193 WDYT?